### PR TITLE
Add RichHandler suppress

### PR DESCRIPTION
**Pull request by daddycocoaman**
## Type of changes

- [ ] Bug fix
- [x] New feature
- [x] Documentation / docstrings
- [ ] Tests
- [ ] Other

## Checklist

- [x] I've run the latest [black](https://github.com/psf/black) with default args on new code.
- [x] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [ ] I've added tests for new code.
- [x] I accept that @willmcgugan may be pedantic in the code review.

## Description

This PR address #1836 and adds the `suppress` functionality for Traceback to Rich Handler as well. The new `tracebacks_suppress` parameter is passed to the `Traceback` object created when `rich_tracebacks=True`.

One issue I had was creating a test. The test for Traceback suppress only counts the length of `suppress` after the modules/paths resolves. For this test, I tried to `emit` a custom `LogRecord` but I think the way Traceback handles rendering suppressed messages isn't compatible with an non-existent file. In other words, to ensure `RichHandler(console=console, rich_tracebacks=True, tracebacks_suppress=["foo.bar"])` is properly tested, an actual `foo` file should exist but I'm not sure how you want to handle extra files in the `tests` folder. I also tried supressing `pytest` but failed at that too.

### CODE REVIEW

File: docs/source/logging.rst [hunk 1]
```diff
@@ -49,3 +49,23 @@ The :class:`~rich.logging.RichHandler` class may be configured to use Rich's :cl
...
+
+    import click
+    import logging
+    from rich.logging import RichHandler
+
+    logging.basicConfig(
+        level="NOTSET",
+        format="%(message)s",
+        datefmt="[%X]",
+        handlers=[RichHandler(rich_tracebacks=True, tracebacks_suppress)]
```
Comment (willmcgugan) [inline docs/source/logging.rst:68]: Did you mean?

```python
tracebacks_suppress=[click]
```

File: docs/source/logging.rst [hunk 2]
```diff
@@ -49,3 +49,23 @@ The :class:`~rich.logging.RichHandler` class may be configured to use Rich's :cl
 
 
 There are a number of other options you can use to configure logging output, see the :class:`~rich.logging.RichHandler` reference for details.
+
+Suppressing Frames
+------------------
+
+If you are working with a framework (click, django etc), you may only be interested in seeing the code from your own application within the traceback. You can exclude framework code by setting the `suppress` argument on `Traceback`, `install`, and `Console.print_exception`, which should be a list of modules or str paths.
```
Comment (willmcgugan) [inline docs/source/logging.rst:56]: Could you update this to refer to the RichHandler.

**Review by willmcgugan — CHANGES_REQUESTED** [review]
I'm happy with not writing an explicit test for this. It's such a small surface area for any faults.

Just a couple requests re doc.

**Review by willmcgugan — APPROVED** [review]

**Comment by willmcgugan** [comment]
I think black was updated since you did this. Would you mind reformatting? And then should be good to go.

**Comment by daddycocoaman** [comment]
> I think black was updated since you did this. Would you mind reformatting? And then should be good to go.

Sorry for the delay...haven't been paying attention to my GH feed lately. I've updated as requested.

**Comment by willmcgugan** [comment]
Thanks!

**Pull request merged**